src/HttpClient.php

```diff
@@ -88,24 +88,69 @@ protected function initializeHandle()
     protected function getBody(RequestInterface $request)
     {
         if (empty($request->getParameters())) {
-            return $request->getBody();
+            return [ "data" => $request->getBody(), "headers" => [] ];
         }
         // special handling for form parameters i.e.
         // returning flatten array with encoded keys if any multipart parameter exists
         // OR returning concatenated encoded parameters string
         $encodedBody = join('&', $request->getEncodedParameters());
         $multipartParameters = $request->getMultipartParameters();
         if (empty($multipartParameters)) {
-            return $encodedBody;
+            return [ 'data' => $encodedBody, 'headers' => [ 'content-type' => 'application/x-www-form-urlencoded' ] ];
         }
         if (empty($encodedBody)) {
-            return $multipartParameters;
+            return $this->createMultiPartFormData($multipartParameters);
         }
         foreach (explode('&', $encodedBody) as $param) {
             $keyValue = explode('=', $param);
             $multipartParameters[urldecode($keyValue[0])] = urldecode($keyValue[1]);
         }
-        return $multipartParameters;
+        return $this->createMultiPartFormData($multipartParameters);
+    }
+
+    private function createMultiPartFormData($multipartParameters)
+    {
+        $i = 1;
+        $length = count($multipartParameters);
+        $boundary = uniqid();
+        $post_data = '';
+
+        foreach ($multipartParameters as $key => $parameter) {
+            $post_data .= '--' . $boundary . "\r\n";
+            if ($parameter instanceof \CURLFile) {
+                $post_data .= 'Content-Disposition: form-data; name="'
+                    . $key
+                    . '"; filename="' . basename($parameter->getFilename())
+                    . '"' . "\r\n";
+
+                $post_data .= 'Content-Type: ' . $parameter->getMimeType() . "\r\n\r\n";
+                $post_data .= file_get_contents($parameter->getFilename()) . "\r\n";
+            } elseif (isset($parameter["headers"])) {
+                $post_data .= 'Content-Disposition: form-data; name="' . $key . '"' . "\r\n";
+                foreach($parameter["headers"] as $headerKey => $headerValue) {
+                    $post_data .= $headerKey . ': ' . $headerValue . "\r\n\r\n";
```

Comment:
This line adds additional `\r\n` after each header kindly remove extra line between each header